src/corefx/System.Globalization.Native/pal_localeStringData.c

```diff
@@ -58,16 +19,17 @@ UErrorCode
 GetLocaleInfoDecimalFormatSymbol(const char* locale, UNumberFormatSymbol symbol, UChar* value, int32_t valueLength)
 {
     UErrorCode status = U_ZERO_ERROR;
-    UNumberFormat* pFormat = unum_open(UNUM_DECIMAL, nullptr, 0, locale, nullptr, &status);
-    UNumberFormatHolder formatHolder(pFormat, status);
+    UNumberFormat* pFormat = unum_open(UNUM_DECIMAL, NULL, 0, locale, NULL, &status);
 
     if (U_FAILURE(status))
     {
+        unum_close(pFormat);
```

Comment:
There is nothing to close when the API fails. 

If you read the ICU docs and samples, their preferred style of error handling actually looks like this:

```
UErrorCode status = U_ZERO_ERROR;
UNumberFormat* pFormat = unum_open(UNUM_DECIMAL, NULL, 0, locale, NULL, &status);
unum_getSymbol(pFormat, symbol, value, valueLength, &status);
unum_close(pFormat);
return status;
```

It works because the first thing that every method does is that will check whether status is still no-error. If it is error, it immediately returns. It results into more compact code without superfluous error handling.

Similar in other cases.

Reply:
OK.

> superfluous error handling

Isn't this the case in current master where RAII pattern is implemented with the help of this header https://github.com/dotnet/coreclr/blob/master/src/corefx/System.Globalization.Native/holders.h#L32-L86 (this PR removes and replaces with the calls inline)?

Reply:
The holder constructor checks for error here https://github.com/dotnet/coreclr/blob/master/src/corefx/System.Globalization.Native/holders.h#L15 . And the destructor a few lines later does nothing when the pointer is null.